src/main/scala/ch/epfl/lamp/CourseraBuild.scala

```diff
@@ -126,35 +123,20 @@ trait CourseraBuild extends sbt.Build {
   val grade = TaskKey[Unit]("grade")
 
   // mapR: submit the grade / feedback in any case, also on failure
-  lazy val gradeSetting = grade <<= (gradingUUID, scalaTestSubmission, styleCheckSubmission, apiKey, gradeProjectDetails, streams) mapR { (uuidR, sts, scs, apiKeyR, projectDetailsR, s) =>
-    val Value(uuid) = uuidR
+  lazy val gradeSetting = grade <<= (scalaTestSubmission, styleCheckSubmission, gradeProjectDetails, streams) mapR { (sts, scs, projectDetailsR, s) =>
     val logOpt = s match {
       case Value(v) => Some(v.log)
       case _ => None
     }
-    logOpt.foreach(_.info(GradingFeedback.feedbackString(uuid, html = false)))
+    val feedback = GradingFeedback.feedbackString(html = false)
+    // Output to logs for instructor debugging purposes.
+    logOpt.foreach(_.info(feedback))
     val Value(projectDetails) = projectDetailsR
-    apiKeyR match {
-      case Value(apiKey) if (!apiKey.isEmpty) =>
-        logOpt.foreach(_.debug("Course Id for submission: " + projectDetails.courseId))
-        logOpt.foreach(_.debug("Corresponding API key: " + apiKey))
-        // if build failed early, we did not even get the api key from the submission queue
-        if (!GradingFeedback.apiState.isEmpty && !Settings.offlineMode) {
-          val scoreString = "%.2f".format(GradingFeedback.totalScore)
-          CourseraHttp.submitGrade(GradingFeedback.feedbackString(uuid), scoreString, GradingFeedback.apiState, apiKey, projectDetails, logOpt) match {
-            case Failure(msgs) =>
-              sys.error(msgs.list.mkString("\n"))
-            case _ =>
-              ()
-          }
-        } else if (Settings.offlineMode) {
-          logOpt.foreach(_.info(" \nSettings.offlineMode enabled, not uploading the feedback"))
-        } else {
-          sys.error("Could not submit feedback - apiState not initialized")
-        }
-      case _ =>
-        sys.error("Could not submit feedback - apiKey not defined: " + apiKeyR)
-    }
+    val didPass = GradingFeedback.totalScore == (GradingFeedback.maxTestScore + GradingFeedback.maxStyleScore)
+    val gradeOutputJson = JsObject(("isCorrect" -> JsBoolean(didPass)), ("feedback" -> JsString(feedback)))
+    // Output grade to stderr because stdout is polluted. Swap grade output to stdout in the wrapper script.
+    System.err.println(gradeOutputJson.compactPrint)
```

Comment:
Why not have a file (e.g., in `/shared`) where we store the student output. Libraries use `System.err` for all kinds of logging and students will get all of this although it might not be desired.